common/config/secret/secret_parser.go

```diff
@@ -23,6 +24,38 @@ var DecodeHook mapstructure.DecodeHookFunc = func(from reflect.Type, to reflect.
 		return data, nil
 	}
 
+	// Check if target type is a slice of pointers to Secret
+	if to.Kind() == reflect.Slice {
+		// Check if the slice element is a pointer to Secret
+		if to.Elem().Kind() == reflect.Ptr && to.Elem().Elem() == reflect.TypeOf((*Secret)(nil)).Elem() {
+			// Get the source data as a string
+			var sourceStr string
+			switch v := data.(type) {
+			case string:
+				sourceStr = v
+			case []byte:
+				sourceStr = string(v)
+			default:
+				// If it's not a string or []byte, let mapstructure handle it normally
+				return data, nil
```

Comment:
If you got to this point, that means this _was_ meant to be a secret. But in this `default` case, you are just returning the `data` without hiding it within a `Secret` struct.

I don't fully understand what edge case would cause this to happen, but this doesn't quite seem right to me.